### PR TITLE
Add support for Power8

### PR DESCRIPTION
**Pull request by niroyb**
@Microsoft/ostc-devs

### CODE REVIEW

File: Unix/buildtool [hunk 1]
```diff
@@ -514,7 +518,7 @@ if [ "$arg1" = "cc" ]; then
 
     case "$platform" in
 
-        LINUX_IX86_GNU|LINUX_X86_64_GNU)
+        LINUX_*)
```
Comment (jeffaco) [inline Unix/buildtool]: I'm not sure it's a good idea for this to be so "open". Why not just add the platform for Linux on PPC to this list? I think that's safer, FWIW.

File: Unix/buildtool [hunk 2]
```diff
@@ -555,7 +559,7 @@ if [ "$arg1" = "cxx" ]; then
     fi
 
     case "$platform" in
-        LINUX_IX86_GNU|LINUX_X86_64_GNU)
+        LINUX_*)
```
Comment (jeffaco) [inline Unix/buildtool]: See above.

File: Unix/buildtool [hunk 3]
```diff
@@ -643,7 +647,7 @@ if [ "$arg1" = "size" ]; then
 
             echo "echo \'size not avaliable\'"
             ;;
-        AIX_PPC_IBM)
+        AIX_PPC_IBM|LINUX_PPC_GNU)
```
Comment (jeffaco) [inline Unix/buildtool]: Ah, that's better. Can't you use these above as well?

File: Unix/buildtool [hunk 4]
```diff
@@ -706,7 +714,7 @@ if [ "$arg1" = "cflags" -o "$arg1" = "cxxflags" ]; then
     r=""
 
     case "$platform" in
-        LINUX_IX86_GNU|LINUX_X86_64_GNU|MONTAVISTA_IX86_GNU|NETBSD_IX86_GNU)
+        LINUX_*|MONTAVISTA_IX86_GNU|NETBSD_IX86_GNU)
```
Comment (jeffaco) [inline Unix/buildtool]: So ditto for lines 717, 891, 1015, 1099, 1143, and a bunch of others. I won't mention the others, you can find them.

File: Unix/buildtool [hunk 5]
```diff
@@ -725,8 +733,13 @@ if [ "$arg1" = "cflags" -o "$arg1" = "cxxflags" ]; then
             r="$r -fno-strict-aliasing"
             r="$r -D_GNU_SOURCE"
             r="$r -D_XOPEN_SOURCE=600"
-            r="$r -D_BSD_SOURCE"
 
+	    # _BSD_SOURCE is deprecated on PPC
+            if test "$platform" = "LINUX_PPC_GNU"; then
+		r="$r -DEFAULT_SPOURCE"
```
Comment (jeffaco) [inline Unix/buildtool]: DEFAULT_SPORCE? Should this be DEFAULT_SOURCE?

File: Unix/configure [hunk 6]
```diff
@@ -1789,7 +1789,7 @@ fi
 
 cat $root/common/MI.h >> $fn
 
-echo "Created $fn"
+echo "created $fn"
```
Comment (jeffaco) [inline Unix/configure]: Why this change?

File: Unix/sock/sock.c [hunk 7]
```diff
@@ -361,6 +361,7 @@ MI_Result Sock_Connect(
 
     r = connect(self, &addr->u.sock_addr, (size_t)addr->sock_addr_size);
     connectErrno = errno;
+    (void) connectErrno;
```
Comment (jeffaco) [inline Unix/sock/sock.c]: connectErrno isn't being used? Then why not get rid of line 363?

File: Unix/configure (hunk 6 above)
Comment (niroyb) [inline Unix/configure]: All the others are lowercase in this file.

File: Unix/sock/sock.c (hunk 7 above)
Comment (niroyb) [inline Unix/sock/sock.c]: It is used if debug logging is compiled in. Line 375

Comment (jeffaco) [inline Unix/sock/sock.c]: Doesn't this need to be #ifdef'ed for DEBUG or something? I thought that's what we discussed yesterday ...
